packages/profile-sync-controller/src/controllers/user-storage/network-syncing/controller-integration.ts

```diff
@@ -0,0 +1,69 @@
+import log from 'loglevel';
+
+import type { UserStorageBaseOptions } from '../services';
+import type { UserStorageControllerMessenger } from '../UserStorageController';
+import { addNetwork, deleteNetwork, updateNetwork } from './sync';
+
+type SetupNetworkSyncingProps = {
+  messenger: UserStorageControllerMessenger;
+  getStorageConfig: () => Promise<UserStorageBaseOptions>;
+};
+
+/**
+ * Initialize and setup events to listen to for network syncing
+ * @param props - parameters used for initializing and enabling network syncing
+ */
+export function startNetworkSyncing(props: SetupNetworkSyncingProps) {
+  const { messenger, getStorageConfig } = props;
+
+  try {
+    messenger.subscribe(
+      'NetworkController:networkAdded',
+      // eslint-disable-next-line @typescript-eslint/no-misused-promises
+      async (networkConfiguration) => {
+        try {
+          const opts = await getStorageConfig();
+          await addNetwork(networkConfiguration, opts);
+        } catch {
+          // Silently fail sync
+        }
+      },
+    );
+  } catch (e) {
+    log.warn('NetworkSyncing, event subscription failed', e);
```

Comment:
Something I've noticed. If the controller does not have the allowedEvent set and we  try to start listening, it will throw an error.

Although this is a rare edge case, just wanted to add these checks so the controller doesn't blow up because it is missing an event.